lib/palindrome_check.rb

```diff
@@ -1,5 +1,33 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
+# Time complexity is O(n/2) --> O(n)
+# Space complexity is O(1) as the check is being done in place
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+
+  index = 0
+  first_index = 0
+  last_index = my_phrase.length - 1
+  check = true
+
+  while index <= my_phrase.length/2
+    # Accounts for spaces for first half of string
+    while my_phrase[first_index + index] == " "
+      first_index += 1
+    end
+
+    #Accounts for spaces for second half of string
+    while my_phrase[last_index - index] == " "
+      last_index -= 1
+    end
+
+    # palindrome check
+    if my_phrase[first_index + index] != my_phrase[last_index - index]
+      check = false
```

Comment:
If the first and last character itself do not match, you needn't compare the remaining characters by continuing in the loop. Hence, instead of assigning `false` value to `check`, I'd suggest returning false on line 28.

Reply:
Accordingly, line 32 could simply be `return true`.